pandas/tests/base/test_conversion.py

```diff
@@ -406,13 +401,13 @@ def test_to_numpy_copy(arr, as_series, using_infer_string):
 
     # no copy by default
     result = obj.to_numpy()
-    if using_infer_string and arr.dtype == object:
+    if using_infer_string and arr.dtype == object and obj.dtype.storage == "pyarrow":
         assert np.shares_memory(arr, result) is False
     else:
         assert np.shares_memory(arr, result) is True
 
     result = obj.to_numpy(copy=False)
-    if using_infer_string and arr.dtype == object:
+    if using_infer_string and arr.dtype == object and obj.dtype.storage == "pyarrow":
```

Comment:
Why if a user specifically requests not to make a copy are we converting the numpy array to a pyarrow arrow baked string array for what is an immutable index? There are other performance benefits?

Reply:
It's the other way around, the pyarrow array (stored under the hood in `obj`) is being converted to a numpy array, and that can just never be done without a copy (different memory layout)

Reply:
the first line of the test is `obj = pd.Index(arr, copy=False)`

so if we have a numpy `arr` and specify `copy=False` for the immutable index we get a pyarrow backed index and a copy is made? and then the `.to_numpy()` method makes another copy?

```python
>>> pd.options.future.infer_string = True
>>> arr = np.array(["a", "b", "c"], dtype=object)
>>> arr
array(['a', 'b', 'c'], dtype=object)
>>> 
>>> idx = pd.Index(arr, copy=False)
>>> idx
Index(['a', 'b', 'c'], dtype='str')
>>> 
```

so the question is perhaps should `idx = pd.Index(arr, copy=False)` return an `Index` with a string dtype, perhaps raise like numpy now do for `__array__` when a copy can't be made or is this a moot point when CoW is extended to the Indexes as the copy keyword would be irrelevant?